nilearn/decoding/space_net.py

```diff
@@ -0,0 +1,1247 @@
+"""
+sklearn-compatible implementation of spatially structured learners (
+TV-L1, S-LASSO, etc.)
+
+"""
+# Author: DOHMATOB Elvis Dopgima,
+#         PIZARRO Gaspar,
+#         VAROQUAUX Gael,
+#         GRAMFORT Alexandre,
+#         EICKENBERG Michael,
+#         THIRION Bertrand
+# License: simplified BSD
+
+import warnings
+import numbers
+import time
+import sys
+from functools import partial
+import numpy as np
+from scipy import stats, ndimage
+from sklearn.base import RegressorMixin, clone
+from sklearn.utils.extmath import safe_sparse_dot
+from sklearn.linear_model.base import LinearModel, center_data
+from sklearn.feature_selection import (SelectPercentile, f_regression,                                       f_classif)
+from sklearn.externals.joblib import Memory, Parallel, delayed
+from sklearn.cross_validation import check_cv
+from .._utils.compat import _basestring
+from ..input_data import NiftiMasker
+from .._utils.fixes import LabelBinarizer, atleast2d_or_csr
+from .objective_functions import _unmask
+from .space_net_solvers import (tvl1_solver, smooth_lasso_logistic,
+                                smooth_lasso_squared_loss)
+
+
+# Volume of a standard (MNI152) brain mask in mm^3
+MNI152_BRAIN_VOLUME = 1827243.
+
+
+def _get_mask_volume(mask_img):
+    """Computes the volume of a brain mask in mm^3
+
+    Parameters
+    ----------
+    mask_img : nibabel image object
+        Input image whose voxel dimensions are to be computed.
+
+    Returns
+    -------
+    vol : float
+        The computed volume.
+    """
+    vox_dims = mask_img.get_header().get_zooms()[:3]
+    return 1. * np.prod(vox_dims) * mask_img.get_data().astype(np.bool).sum()
+
+
+def _crop_mask(mask):
```

Comment:
Ya just seen it. Quite cool. However, it'd be an overkill for what I'm trying to do here (for example, i don't have / need an affine, etc.).